Add PSET Guide (#538)
figment-networks/figment-documentation: docs/staking/04_Guides/presigned-exit-transactions.mdx

---
title: Pre-signed Exit Transactions
sidebar_position: 2
pagination_prev: null
pagination_next: null
---

Exiting a validator on Ethereum requires submission of a pre-signed exit transaction to the network, which requires a signature by the validator's private key (held only by Figment to reduce slashing risk). 

Pre-signed exit transactions provide a safe way to ensure you can exit your Ethereum validators with minimal future dependence
on your staking provider. Once the pre-signed exit transaction is generated, anyone with access can broadcast it to exit the validator,
which is why Figment requires an encryption key, so only you have access to them.

Since pre-signed exit transactions expire after two network hard forks, Figment re-generates them daily at 6:30am UTC.
It is up to you to decide how often to update them in your system.

This guide will walk you through the process of using pre-signed exit transactions with Figment.

## Creating a Keypair

Figment currently supports both **GPG** and **RSA** keys for encrypting pre-signed exit transactions.

> Note that Figment customers must use a _single encryption key_ for all of their validators.

Reviewer: ```suggestion
```

Please contact your Figment Customer Success Representative to provide or update your encryption key(s).

### GPG

Figment requires a GPG key size of at least 2048 bits.

- We recommend using the following command to generate your key: `gpg --full-generate-key`.
- To export your public key (which you will provide to Figment): `gpg --output public_key.gpg --armor --export colton@figment.io`

### RSA

- To generate your private key: `openssl genrsa -out private_key.pem 4096`
  - Note that we require an RSA key size of 4096 bits
- To generate the public key (which you will provide to Figment): `openssl rsa -in private_key.pem -outform PEM -pubout -out public_key.pem`
  - Note that the public key should be in **PKCS8** format, _i.e._ starting with: `----BEGIN PUBLIC KEY-----`

## After providing your key to Figment

You can expect your public key to be saved within 24 business hours of providing it to Figment. Once the public key has been saved,

> an up-to-date version of the encrypted pre-signed exit transaction for all of your active validators will be generated daily at 6:30am UTC.

Reviewer: ```suggestion
an up-to-date version of the encrypted pre-signed exit transaction for all of your **active** validators will be generated daily at 6:30am UTC.
```

---
title: Pre-signed Exit Transactions
sidebar_position: 2
pagination_prev: null
pagination_next: null
---

> Exiting a validator on Ethereum requires submission of a Voluntary Exit on the consensus layer, signed by the validator private key,
> which is always held by Figment to reduce slashing risk.

Reviewer: ```suggestion
Exiting a validator on Ethereum requires submission of a pre-signed exit transaction to the network, which requires a signature by the validator's private key (held only by Figment to reduce slashing risk). 
```

Pre-signed exit transactions provide a safe way to ensure you can exit your Ethereum validators with minimal future dependence

> on your staking provider. Once the Voluntary Exit is generated, anyone can broadcast it on the consensus layer to exit the validator,

Reviewer: ```suggestion
on your staking provider. Once the pre-signed exit transaction is generated, anyone with access can broadcast it to exit the validator,
```

which is why Figment requires an encryption key, so only you have access to them.

> Since pre-signed exit transactions expire after two hard forks on the network, Figment re-generates them every day at 6:30am UTC.

Reviewer: ```suggestion
Since pre-signed exit transactions expire after two network hard forks, Figment re-generates them daily at 6:30am UTC.
```

It is up to you to decide how often to update them in your system.

> This guide will walk you through the process of using pre-signed exit transactions with Figment.

Reviewer: remove this line

Author: @coltonc3 Which line? 😅

Reviewer: ha sorry. pls remove "This guide will walk you through the process of using pre-signed exit transactions with Figment." @pyr0gan

## Creating a Keypair

> Figment currently supports both **GPG** and **RSA** for encrypting pre-signed exit transactions.

Reviewer: ```suggestion
Figment currently supports both **GPG** and **RSA** keys for encrypting pre-signed exit transactions.
```
